controllers/complianceeventsapi/server.go

```diff
@@ -749,6 +802,27 @@ func postComplianceEvent(db *sql.DB, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	allowed, err := canRecordComplianceEvent(cfg, authenticatedClient, authenticator, reqEvent.Cluster.Name, r)
+	if err != nil {
+		if errors.Is(err, ErrNotAuthorized) {
+			writeErrMsgJSON(w, "Unauthorized", http.StatusUnauthorized)
```

Comment:
This error is propagator SA doesn't have "`CREATE`" for SubjectAccessReviews.  I think it is `StatusInternalServerError`.

Reply:
```
There's a problem with 401 Unauthorized, the HTTP status code for authentication errors. And that’s just it: it’s for authentication, not authorization. Receiving a 401 response is the server telling you, “you aren’t authenticated–either not authenticated at all or authenticated incorrectly–but please reauthenticate and try again.” To help you out, it will always include a WWW-Authenticate header that describes how to authenticate.
```
This is what I found. Here isn't about `authentication`. Is it?!!!

Reply:
I think it's right. If the user provided credential (token or cert) does not allow us to get to the subject access review call, then it's unauthorized (bad authentication provided). If the subject access review says the user is not allowed then it should be a forbidden status code.

Reply:
Also you already wrote `//+kubebuilder:rbac:groups=authorization.k8s.io,resources=subjectaccessreviews,verbs=create` This won't fall into here